src/PrestaShopBundle/Controller/Api/Improve/Design/PositionsController.php

```diff
@@ -0,0 +1,182 @@
+                ],
+                $request
+            );
+        }
+
+        if (!$module->updatePosition($hookId, $way, $position)) {
+            return $this->jsonResponse(
+                [
+                    'hasError' => true,
+                    'errors' => ['Cannot update module position.'],
+                ],
+                $request
+            );
+        }
+
+        return $this->jsonResponse([], $request);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getHookableListAction(Request $request)
+    {
+        if (_PS_MODE_DEMO_) {
+            return $this->jsonResponse(
+                [
+                    'hasError' => true,
+                    'errors' => ['Live Edit: This functionality has been disabled.'],
+                ],
+                $request
+            );
+        }
+
+        $modules = $request->request->get('modules');
+        $hooks = $request->request->get('hooks');
+        if (empty($hooks) or empty($modules)) {
+            return $this->jsonResponse(
+                [
+                    'hasError' => true,
+                    'errors' => ['Live Edit: no module on this page.'],
+                ],
+                $request
+            );
+        }
+
+        $hookableList = [];
+        foreach ($modules as $module) {
+            $module = trim($module);
+            if (empty($module)) {
+                continue;
+            }
+
+            if (!Validate::isModuleName($module)) {
+                return $this->jsonResponse(
+                    [
+                        'hasError' => true,
+                        'errors' => ['Live Edit: module is invalid.'],
+                    ],
+                    $request
+                );
+            }
+
+            $moduleInstance = Module::getInstanceByName($module);
+            foreach ($hooks as $hookName) {
+                $hookName = trim($hookName);
+                if (empty($hookName)) {
+                    continue;
+                }
+
+                if (!isset($hookableList[$hookName])) {
+                    $hookableList[$hookName] = [];
+                }
+
+                if ($moduleInstance->isHookableOn($hookName)) {
+                    $hookableList[$hookName][] = str_replace('_', '-', $module);
+                }
+            }
+        }
+
+        return $this->jsonResponse($hookableList, $request);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getHookableModuleListAction(Request $request)
+    {
+        if (_PS_MODE_DEMO_) {
+            return $this->jsonResponse(
+                [
+                    'hasError' => true,
+                    'errors' => ['Live Edit: This functionality has been disabled.'],
+                ],
+                $request
+            );
+        }
+
+        $hookName = $request->request->get('hook');
```

Comment:
same here, and even in adapters, you must avoid to use legacy classes directly but instead the adapters.